packages/api-server/rest_server/app.py

```diff
@@ -0,0 +1,51 @@
+import asyncio
+import logging
+import os
+import sys
+
+import socketio
+from fastapi import FastAPI
+
+from .app_config import app_config
+from .building_map import building_map_router
+
+logger = logging.getLogger("rest_app")
+handler = logging.StreamHandler(sys.stdout)
+handler.setFormatter(logging.Formatter(logging.BASIC_FORMAT))
+logger.addHandler(handler)
+if "RMF_REST_SERVER_DEBUG" in os.environ:
+    logger.setLevel(logging.DEBUG)
+else:
+    logger.setLevel(logging.INFO)
+
+app = FastAPI()
+sio = socketio.AsyncClient()
+
+
+@app.on_event("startup")
+async def on_startup():
+    await sio.connect(app_config.api_server_url)
+    connected = asyncio.Future()
+    sio.on("connect", lambda: connected.set_result(True))
+    await connected
+    del sio.handlers["/"]["connect"]
+
+    app.include_router(
+        building_map_router(sio, logger.getChild("building_map")),
+        prefix="/building_map",
+    )
+
+    logger.info("started app")
+
+
+# There is a big with uvicorn and socketio that causes
```

Comment:
big -> bug . I think your solution is super valid.

Reply:
fyi, the "solution" I did is to simply split them up into 2 ASGI app, because of the bug, we can't run fastapi and socketio together (we could, but need handle cleanup of both apps manually). So the easier solution is to simply just split them up, its also ends up being more "micro-service".